chainercv/experimental/links/model/fcis/fcis.py

```diff
@@ -0,0 +1,308 @@
+            self.limit = 100
+            self.min_drop_size = 16
+        elif preset == 'evaluate':
+            self.nms_thresh = 0.3
+            self.score_thresh = 1e-3
+            self.mask_merge_thresh = 0.5
+            self.binary_thresh = 0.4
+            self.limit = 100
+            self.min_drop_size = 16
+        else:
+            raise ValueError('preset must be visualize or evaluate')
+
+    def predict(self, imgs):
+        """Segment object instances from images.
+
+        This method predicts instance-aware object region for each image.
+
+        Args:
+            imgs (iterable of numpy.ndarray): Arrays holding images of shape
+                :math:`(B, C, H, W)`.  All images are in CHW and RGB format
+                and the range of their value is :math:`[0, 255]`.
+
+        Returns:
+           tuple of lists:
+           This method returns a tuple of three lists,
+           :obj:`(masks, labels, scores)`.
+
+           * **masks**: A list of float arrays of shape :math:`(R, H, W)`, \
+               where :math:`R` is the number of masks in a image. \
+               Each pixel holds value if it is inside the object inside or not.
+           * **labels** : A list of integer arrays of shape :math:`(R,)`. \
+               Each value indicates the class of the masks. \
+               Values are in range :math:`[0, L - 1]`, where :math:`L` is the \
+               number of the foreground classes.
+           * **scores** : A list of float arrays of shape :math:`(R,)`. \
+               Each value indicates how confident the prediction is.
+
+        """
+
+        prepared_imgs = []
+        sizes = []
+        for img in imgs:
+            size = img.shape[1:]
+            img = self.prepare(img.astype(np.float32))
+            prepared_imgs.append(img)
+            sizes.append(size)
+
+        masks = []
+        labels = []
+        scores = []
+
+        for img, size in zip(prepared_imgs, sizes):
+            with chainer.using_config('train', False), \
+                    chainer.function.no_backprop_mode():
+                # inference
+                img_var = chainer.Variable(self.xp.array(img[None]))
+                scale = img_var.shape[3] / size[1]
+                roi_seg_scores, _, roi_scores, rois, _ = \
+                    self.__call__(img_var, scale)
+
+            roi_seg_score = roi_seg_scores.array
+            roi_score = roi_scores.array
+            bbox = rois / scale
+
+            # shape: (n_rois, 4)
+            bbox[:, 0::2] = self.xp.clip(bbox[:, 0::2], 0, size[0])
+            bbox[:, 1::2] = self.xp.clip(bbox[:, 1::2], 0, size[1])
+
+            roi_seg_prob = F.softmax(roi_seg_score).array
+            roi_prob = F.softmax(roi_score).array
+
+            roi_seg_prob = chainer.cuda.to_cpu(roi_seg_prob)
+            roi_prob = chainer.cuda.to_cpu(roi_prob)
+            bbox = chainer.cuda.to_cpu(bbox)
+
+            roi_mask_score, bbox, label, score = mask_voting(
```

Comment:
Isn't the returned mask a probability?
In that case, can you rename `roi_mask_score` to `roi_mask_prob`.